syne_tune/config_space.py

```diff
@@ -605,6 +605,99 @@ def __eq__(self, other) -> bool:
         )
 
 
+class OrdinalNearestNeighbor(Ordinal):
```

Comment:
why does the sampling logic change? in what I've envisioned this would be just as `Ordinal` with the difference that the preprocessing for surrogate models etc is done differently. If ordinal has only types floats/ints, they will be preprocessed as floats/ints (potentially with log scale). If not, standard behavior for Ordinal. Breaking it down into too many different search spaces will make it very complex.

Reply:
Well, the point about "nn" and "nn-log" is they are finite-size versions of uniform and loguniform, which implies a non-uniform distribution over the values.

When you do a encoding to [0, 1], it is important the random sampling is the same as sampling uniformly from encoding and mapping back.